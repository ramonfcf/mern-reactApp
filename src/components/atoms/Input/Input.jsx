import propTypes from "prop-types";

const Input = ({ type, value, onChange, label, placeholder }) => {
  return (
    <div className="form-group py-2">
      <label htmlFor={label}>{label}</label>
      <input
        type={type}
        className="form-control"
        id={label}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default Input;

Input.propTypes = {
  type: propTypes.string.isRequired,
  value: propTypes.string.isRequired,
  onChange: propTypes.func.isRequired,
  label: propTypes.string.isRequired,
  placeholder: propTypes.string.isRequired
};
