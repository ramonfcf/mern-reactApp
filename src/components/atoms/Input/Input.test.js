import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

import Input from "./Input";

describe("Input component", () => {
  const mockOnChange = jest.fn();
  const label = "Test label";
  const placeholder = "Test placeholder";

  it("matches snapshot", () => {
    const { container } = render(
      <Input
        type="text"
        value="Test value"
        onChange={mockOnChange}
        label={label}
        placeholder={placeholder}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("calls onChange prop when input value changes", () => {
    render(
      <Input
        type="text"
        value="Test value"
        onChange={mockOnChange}
        label={label}
        placeholder={placeholder}
      />
    );
    const inputElement = screen.getByRole("textbox");

    fireEvent.change(inputElement, { target: { value: "New value" } });
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });

  it("renders with correct label", () => {
    render(
      <Input
        type="text"
        value="Test value"
        onChange={mockOnChange}
        label={label}
        placeholder={placeholder}
      />
    );
    const labelElement = screen.getByText(label);

    expect(labelElement).toHaveTextContent(label);
  });

  it("renders with correct placeholder", () => {
    render(
      <Input
        type="text"
        value="Test value"
        onChange={mockOnChange}
        label={label}
        placeholder={placeholder}
      />
    );
    const inputElement = screen.getByRole("textbox");

    expect(inputElement).toHaveAttribute("placeholder", placeholder);
  });
});
