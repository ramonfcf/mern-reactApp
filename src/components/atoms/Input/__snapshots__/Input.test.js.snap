// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component matches snapshot 1`] = `
<div>
  <div
    class="form-group py-2"
  >
    <label
      for="Test label"
    >
      Test label
    </label>
    <input
      class="form-control"
      id="Test label"
      placeholder="Test placeholder"
      type="text"
      value="Test value"
    />
  </div>
</div>
`;
